assets/javascripts/discourse/components/canned-reply.js.es6

```diff
@@ -15,18 +14,35 @@ export default Ember.Component.extend({
       // TODO: This is ugly. There _must_ be another way.
       // TODO: This code is also duplicated (see controller).
       const composer = Discourse.__container__.lookup("controller:composer");
+      const model = composer.model;
 
-      if (composer.model) {
-        this.appEvents.trigger(
-          "composer:insert-block",
-          this.get("reply.content")
-        );
+      let replyTitle = this.get("reply.title");
+      let replyContent = this.get("reply.content");
 
-        if (!composer.model.get("title")) {
-          composer.model.set("title", this.get("reply.title"));
+      // Replace variables with values.
+      if (model) {
+        const vars = {
```

Comment:
Where are these variables coming from/being used?

Reply:
This is a new feature that has been requested which allows replacing variables like #{user} with the username of current user.

The variables are currently "hardcoded", but it is a feature that has been requested for long time and the implementation was not difficult.

Reply:
Can you link to the conversion(s) on meta?